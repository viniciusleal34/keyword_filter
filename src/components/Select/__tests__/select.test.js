/* eslint-disable no-unused-vars */
import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';

import userEvent from '@testing-library/user-event';
import {Select} from '../index';

describe('Select component', () => {
    const handleEvent = jest.fn();
    const mockOptions = [
        { value: 'Cat', label: 'Meow', title: "Cat" },
        { value: 'Dog', label: 'Woof',  title: "Dog" }
    ];

    it('should render correctly', () => {
        const { container } = render(
            <Select
                placeholder="Select.."
                value=""
                options={mockOptions}
                onChange={handleEvent}
            />
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    it('should not render options', () => {
        expect(screen.queryByText('Meow')).not.toBeTruthy();
        expect(screen.queryByText('Woof')).not.toBeTruthy();
    });
});