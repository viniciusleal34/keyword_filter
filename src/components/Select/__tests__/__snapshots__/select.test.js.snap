// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render correctly 1`] = `
<bds-select
  data-testid="bds-select"
  disabled="false"
  placeholder="Select.."
  value=""
>
  <bds-select-option
    title-text="Cat"
    value="Cat"
  >
    Meow
  </bds-select-option>
  <bds-select-option
    title-text="Dog"
    value="Dog"
  >
    Woof
  </bds-select-option>
</bds-select>
`;
